### PR TITLE
Add test for non invariant default locale and collation

### PR DESCRIPTION
**Pull request by steveharter**

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please (after Jenkins changes related to setting culture-related env vars)

**Comment by steveharter** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@steveharter, given what we saw today with Jenkins and the various LANG/LC_*/etc. env vars, are there any additional tests we can/should add to validate behavior in such cases in the future?

**Comment by steveharter** [comment]
@stephentoub the test themselves should probably be using Invariant culture explicitly and not assume en-us or other compatible culture for string formatting

**Comment by steveharter** [comment]
FWIW, the default locale if no LANG is set is to use en-US-POSIX which has different default formatting settings than en-US or invariant.

**Comment by steveharter** [comment]
Three potential ways to address:
1) Do nothing
2) Change the tests that were failing to explicitly use Invariant
3) Add a test to verify en-US for Linux as the default locale
4) Change functionality to detect en-US-POSIX from Linux and then return the Invariant locale's culturedata or a copy of it (not sure if this is a valid approach)

**Comment by steveharter** [comment]
@ellismg do you have thoughts on the best way to address. I prefer #2 (change tests to use invariant) and\or #3 (verify en-us in unit test). Thanks

**Comment by steveharter** [comment]
FYI the list of tests that assume en-us or invariant:
UInt16Tests.TestTryParse
UInt32Tests.TestTryParse
Int32Tests.TestTryParse
Int64Tests.TestTryParse
Int16Tests.TestTryParse
UInt64Tests.TestTryParse
StringComparerTests.TestCurrent
System.Collections.Specialized.Tests.OrderedDictionaryTests.PassingEqualityComparers
StringTests.TestLastIndexOf
StringTests.TestEndsWith
StringTests.TestStartsWith(comparisonType: CurrentCultureIgnoreCase, text: "Hello", value: "HEL", expected: True)
StringTests.TestIndexOf
TypeTests.TestGetTypeByName(typeName: "System.Nullable`1[System.Int32]", expectedType: typeof(int?))
TypeTests.TestGetTypeByName(typeName: "System.Int32*", expectedType: typeof(System.Int32*))
TypeTests.TestGetTypeByName(typeName: "System.Int32**", expectedType: typeof(System.Int32**))
TypeTests.TestGetTypeByName(typeName: "Outside`1+Inside`1", expectedType: typeof(Outside<,>))
Tests.Expression_Tests.TestCallInstanceMethodsByName
CaseInsensitiveComparer_ctor.ExecuteCaseInsensitiveComparer_ctor
CaseInsensitiveComparer_Default.ExecuteCaseInsensitiveComparer_Default
CaseInsensitiveComparer_Object.ExecuteCaseInsensitiveComparer_Object
CollectionsUtilTests.ExecuteCollectionsUtilTests
System.Reflection.Tests.ModuleTest.GetTypeTest(className: "System.Nullable`1[System.Int32]", expectedType: typeof(int?))
System.Reflection.Tests.ModuleTest.GetTypeTest(className: "System.Int32*", expectedType: typeof(System.Int32*))
System.Reflection.Tests.ModuleTest.GetTypeTest(className: "System.Int32**", expectedType: typeof(System.Int32**))
System.Reflection.Tests.ModuleTest.GetTypeTest(className: "Outside`1+Inside`1", expectedType: typeof(Outside<,>))
System.Reflection.Tests.MethodInfoPropertyTests.TestReturnType2
System.Reflection.Tests.MethodInfoPropertyTests.TestReturnType4
System.Reflection.Tests.ParameterInfoNameTests.TestMethodParams1
System.Reflection.Tests.GetTypeReflectionTests.GetType1

**Comment by ellismg** [comment]
I would prefer if these tests use Invariant.

**Comment by stephentoub** [comment]
Should we merge these test changes, and then address the non-invariant issue separately?

**Comment by steveharter** [comment]
Add issue #3850 to address the tests listed above. I will merge this pull request once I get the OK

File: src/System.Globalization/tests/CultureInfo/CurrentCulture.cs [hunk 1]
```diff
@@ -10,21 +10,40 @@ namespace System.Globalization.Tests
...
+            CultureInfo newUICulture = new CultureInfo(defaultUICulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "ar-SA" : "ja-JP");
             CultureInfo.CurrentUICulture = newUICulture;
             Assert.Equal(CultureInfo.CurrentUICulture, newUICulture);
 
-            CultureInfo.CurrentCulture = defaultCulture;
-            Assert.Equal(CultureInfo.CurrentCulture, defaultCulture);
+            newUICulture = new CultureInfo("de-DE_phoneb");
+            CultureInfo.CurrentUICulture = newUICulture;
+            Assert.Equal(CultureInfo.CurrentUICulture, newUICulture);
+            Assert.Equal("de-DE_phoneb", newUICulture.CompareInfo.Name);
```
Comment (stephentoub) [inline src/System.Globalization/tests/CultureInfo/CurrentCulture.cs]: This is pre-existing, but we should probably be resetting the current culture in a finally block.  That way, if there's a bug that causes one of the asserts to fail, subsequent tests won't be thrown off by the change.

**Comment by stephentoub** [comment]
One pre-existing nit, otherwise LGTM.

**Pull request merged**